src/mainwindow.cpp

```diff
@@ -782,34 +797,44 @@ void MainWindow::on_comboBox_Song_currentTextChanged(const QString &song)
 {
     if (editor && editor->map) {
         editor->map->song = song;
+        editor->map->hasUnsavedDataChanges = true;
```

Comment:
It might be nice to move these two coupled lines into a shared function or similar.